### PR TITLE
Fixes for EMBL beamlines at PETRA

### PR DESCRIPTION
**Pull request by christianbecke**
Processing data recently collected at PETRA P13 using DIALS/XDS failed because the rotation axis was defined incorrectly. There is a new Eiger CdTe detector installed at P14 with serial number E-32-0129 and the Eiger with serial number E-32-0107 was moved to P13. This lead to the P14 rotation axis settings being applied to the P13 data, which resulted in indexing failures.

### CODE REVIEW

**Comment by graeme-winter** [comment]
Started to have a quick look at this now - 

- do you have a couple of example images you could send me so I can be sure that I agree that everything is correct (I am sure it is but I would like to check) 
- do you mind if I take the liberty of adding a `newsfragment` once we have the changes correct
- you are also at liberty to add yourself to the AUTHORS file if you would like, since you are in the most literal sense making yourself an author by contributing code

I will have a read through the change set and make some in-line comments

File: src/dxtbx/format/FormatCBFMiniEigerPetraP13.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            if (
+                "# detector" in record.lower()
+                and "eiger" in record.lower()
+                and "E-32-0107" in record
+            ):
+                return True
+
+        return False
+
+    def _goniometer(self):
```
Comment (graeme-winter) [inline src/dxtbx/format/FormatCBFMiniEigerPetraP13.py:32]: This is functionally the same as the default constructor for goniometer which is implemented in the parent class?

**Review by graeme-winter — COMMENTED** [review]
Small change set, but that is probably in line with what is needed. If the code for P13 and P14 is effectively the same I would keep only one of them and have it capture the serial numbers for both - unless you anticipate some future specialisation which will need different implementations.

Not yet approved as we are lacking a news fragment and I would like to check that the code makes sense by running it with real data.

File: src/dxtbx/format/FormatCBFMiniEigerPetraP13.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+
+from __future__ import annotations
+
+import sys
+
+from dxtbx.format.FormatCBFMiniEiger import FormatCBFMiniEiger
+
+
+class FormatCBFMiniEigerPetraP13(FormatCBFMiniEiger):
```
[the remark this answers is not in the document]
Comment (christianbecke) [reply src/dxtbx/format/FormatCBFMiniEigerPetraP13.py:11]: The code differs in the _goniometer() function, only. Orientation of the rotation axes differ between P14 and P13: at P14, the rotation axis is vertical, while at P13 it is horizontal, i.e. in XDS.INP, we need "ROTATION_AXIS= 0 -1 0" for data from P14 and "ROTATION_AXIS= 1 0 0" for data from P13. So we do need to discriminate between P13 and P14.

File: src/dxtbx/format/FormatCBFMiniEigerPetraP13.py (hunk 1 above)
Comment (christianbecke) [reply src/dxtbx/format/FormatCBFMiniEigerPetraP13.py:32]: Yes, correct. Sorry, I did not check this carefully enough... Looks like we don't need any special code for P13 at all, it works out of the box once the serial number in FormatCBFMiniEigerPetraP14.py is fixed. So [b5e6cdd](https://github.com/cctbx/dxtbx/pull/626/commits/b5e6cddd66990487ff9505bba1c41267dc4e8057) is sufficient to make data from both P13 and P14 work.

**Comment by ndevenish** [comment]
Ah, so if I'm understanding this properly, P13 now works on the plain FormatCBFMiniEiger, so doesn't need a special format class any more.

However, presumably this will break old data being analysed with the detector when it was on the previous beamtime; do you have a changeover date that we can use as a threshold for checking the serial number on the P14 Format class?

**Comment by ndevenish** [comment]
So, I think that this covers the fix and move between beamlines. Thanks to Gleb Bourenkov, we know that the E-32-0107 Eiger moved to P13 on May 22nd 2021. After this date, the standard MiniCBF handling works fine, so we only need to handle P14.

This should now work out what serial is expected on P14 for the timestamp, and checks for that serial.

My only worry is that we don't have any example data files to test this with.

**Comment by graeme-winter** [comment]
@ndevenish anything stopping this from being merged?

**Pull request merged**